helm/external-dns-app/templates/psp.yaml

```diff
@@ -11,7 +11,12 @@ spec:
       - min: {{ .Values.global.securityContext.fsGroupID }}
         max: {{ .Values.global.securityContext.fsGroupID }}
   hostIPC: false
-  hostNetwork: false
+  {{- if .Values.hostNetwork }}
+  hostNetwork: true
+  hostPorts:
+  - min: 7979
+    max: 7979
+  {{- end }}
```

Comment:
if `hostNetwork` isn't set in the `psp`, the value will be always `false` and with `k get psp <yourPSPName> -o yaml` the `hostNetwork` will be omitted. For that reason (and to keep it more readable) if moved everything in the `if` condition.

The technical argument behind `hostNetwork` in CAPZ management clusters:

`hostNetwork` is needed as we don't run `aad-pod-identity` in `CAPZ` management cluster.
For that, requests to the azure-api from the pod-network got blocked by azure itself. To mitigate this, we have to run all the pods in the MC running in `hostNetwork` if the component talks to azure api.

This will change in the future, but we have to implement an additional operator first.

Reply:
Ah, ok. Then it's the other way around: While running in `hostNetwork` you might not need to define `hostPorts` because your container ports already get to be `hostPorts`.